test-crawler/collector/collector.go

```diff
@@ -126,3 +129,29 @@ func ListGoFilesInFolder(root string, ignore []string) (system string, files []s
 
 	return "", nil, nil
 }
+
+// RegenerateName regenerates name adding underscore and usecase index
+// ex: name: ListenProofRequest -> LISTEN_PROOF_REQUEST_001
+func (f *FunctionAnnotation) RegenerateName() {
+	usecase := "001" // in the future it will be autoincremented depending how many return cases we have.
+	buf := &bytes.Buffer{}
+	for i, rune := range f.Name {
+		if unicode.IsUpper(rune) && i > 0 {
+			buf.WriteRune('_')
+		}
+		buf.WriteRune(rune)
+	}
+
+	f.Name = fmt.Sprintf("%s_%s", strings.ToUpper(buf.String()), usecase)
+}
+
+// GenerateDescription generates description based on inputParams, returnParams
+func (f *FunctionAnnotation) RegenerateDescription() {
```

Comment:
See above: return the new value of the description.
Also, we should return a default value when input params / return values are not set. Something like "Function description not set."